core/domain/email_services.py

```diff
@@ -120,17 +120,12 @@ def is_sender_email_valid(sender_email):
     Returns:
         bool. Whether the sender_email is valid.
     """
-    # Checks the case where sender_email is of the form 'Jane <EMAIL>'.
-    sender_email_with_only_first_name_regex = (
-        r'^[a-zA-Z._]+ <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|[0-9]{1,3})' +
-        '(]?)>$')
-    # Checks the case where sender_email is of the form 'Jane Doe <EMAIL>'.
-    sender_email_with_full_name_regex = (
-        r'^[a-zA-Z._]+ [a-zA-Z._]+ <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}' +
-        r'|[0-9]{1,3})(]?)>$')
-    return is_email_valid(sender_email) or (
-        re.search(sender_email_with_only_first_name_regex, sender_email)) or (
-            re.search(sender_email_with_full_name_regex, sender_email))
+    # Regex for 'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or 'email_address'.
+    sender_email_regex = (
+        r'^[a-zA-Z._][a-zA-Z._ ]* <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|' +
```

Comment:
I'm not sure I quite follow what this regex is checking. How was it derived and does this match the checks that were there previously?

> Done, in actuality the original implementation didn't really check for the whether the email or name existed but since that didn't really make sense with the method name, I added this check. Fixed now.

I'm not sure what you mean by "existed". Could you elaborate?

Reply:
Originally for mailgun nothing was checked. This sender_email_valid function did not exist. The only checks that existed for gae_email_services like this
```
    if not _is_email_valid(sender_email):
        return False

    split_sender_email = sender_email.split(' ')
    if len(split_sender_email) < 2:
        return False

    email_address = split_sender_email[-1]
    if not email_address.startswith('<') or not email_address.endswith('>'):
        return False
```
which checked the format but not if the actual email address was valid. The is_email_valid check didn't check for email format either but only checked if the email was a python basestring
```
    if not isinstance(email_address, python_utils.BASESTRING):
        return False

    stripped_address = email_address.strip()
    if not stripped_address:
        return False
```

Since firstly, mailgun allows formats other than the format for _is_sender_email_valid, I decided to combine the two in one regex statement. I also added regex checking to checking the actual email address. 
The regex is basically checking for characters before @and then a site after the @ sign. I tested the regex on a regex checker online.

Reply:
I still don't understand the last part of the regex, though. What does the [0-9]{1,3} represent?

Reply:
So originally, this was supposed to be a way to guarantee there is a valid (site) after the .,
aka .com or .edu or .ch or something. However, now I think it might be too restrictive so I will just make sure there are characters after the final dot.